dashboard/app/models/foorm/form.rb

```diff
@@ -91,6 +98,72 @@ def self.fill_in_library_items(questions)
     return questions
   end
 
+  def validate_questions
+    # fill_in_library_items will throw an exception if any library items are invalid
+    begin
+      filled_questions = Foorm::Form.fill_in_library_items(JSON.parse(questions))
+    rescue StandardError => e
+      errors.add(:questions, e.message)
+      return
+    end
+    filled_questions.deep_symbolize_keys!
```

Comment:
Interesting.  How deep do you expect the hashes to be?  Just pages and then panels with elements under them, at the deepest?

Reply:
Yes that's the deepest normally, although we do support panels of panels (as deep as you want to go). I doubt that will be used much or at all though.